src/sentry/notifications/platform/msteams/provider.py

```diff
@@ -13,11 +19,20 @@
 class MSTeamsRenderer(NotificationRenderer[MSTeamsRenderable]):
     provider_key = NotificationProviderKey.MSTEAMS
 
+    @classmethod
+    def render(cls, *, data: NotificationData, template: NotificationTemplate) -> MSTeamsRenderable:
+        return {}
+
 
 @provider_registry.register(NotificationProviderKey.MSTEAMS)
 class MSTeamsNotificationProvider(NotificationProvider[MSTeamsRenderable]):
     key = NotificationProviderKey.MSTEAMS
     default_renderer = MSTeamsRenderer
+    target_class = IntegrationNotificationTarget
+    target_resource_types = [
+        NotificationTargetResourceType.CHANNEL,
+        NotificationTargetResourceType.DIRECT_MESSAGE,
+    ]
```

Comment:
This concept wasn't in the spec but I realized that it's possible we don't have some of these resources for certain integrations. It's like specifying the feature set of the integration so you know what kind of targets actually work. 

It'll also mean if we build a new integration provider, we don't have to handle channel and DMs at the same time -- for example.

Reply:
I've added it as a check for when the provider validates the target, the target can't specify a resource like "Email" but intend to go to slack (for example)